src/SIM/FunctionSum.C

```diff
@@ -17,14 +17,19 @@
 
 FunctionSum::FunctionSum (FunctionBase* f, double w)
 {
-  comps.push_back(std::make_pair(f,w));
-  ncmp = f->dim();
+  this->add(f,w);
```

Comment:
Consider to inline this constructor in the h-file instead, since now only a singleliner:
```
FunctionSum(FunctionBase* f, double w = 1.0) { this->add(f,w); }
```

Reply:
done.